code/game/objects/objs.dm

```diff
@@ -18,8 +18,10 @@
 	var/max_integrity = 500
 	/// The object will break once obj_integrity reaches this amount in take_damage(). 0 if we have no special broken behavior, otherwise is a percentage of at what point the obj breaks. 0.5 being 50%
 	var/integrity_failure = 0
-	///Damage under this value will be completely ignored
+	/// Damage under this value will be completely ignored
 	var/damage_deflection = 0
+	/// Maximum damage that can be taken in a single hit
+	var/maximum_damage = null
```

Comment:
this var name sounds like how big this object can make damage against others, which fits to attacking objects. It wouldn't be best naming for things like turrets.
`takes_damage_up_to`?

Reply:
takes_damage_up_to is a poor name as it implies the object will stop taking damage after a certain point. I will expand on the name to make it more clear if it wasn't already. I'll go for max_hit_damage since that seems appropriate and chatGPT came up with a similar response when I prompted it for a name suggestion.